include/powtrace.hpp

```diff
@@ -0,0 +1,420 @@
+#pragma once
+#include <vector>
+
+#if (defined(__GNUC__) || defined(__GNUG__)) && \
+    !(defined(__clang__) || defined(__INTEL_COMPILER))
+typedef __float128 qp;
+//#include <quadmath.h>
+#else
+typedef long double qp;
+#endif
+
+namespace libwalrus {
+  enum class StorageType { ColumnOrder, RowOrder };
+
+  template <typename T>
+  inline auto norm_sqr(const std::vector<T> &vec) {
```

Comment:
Don't forget to add in doxygen docstrings to these auxiliary functions! Even though they are small, this helps reduce overhead for future developers. They should also clearly describe each argument and the return type. You can run `make html` from the docs directory to make sure they render correctly in the built docs.

Reply:
You might need to add this file to `docs/conf.py` (where the other hpp files are)